packages/@aws-cdk/aws-s3/lib/rule.ts

```diff
@@ -142,6 +142,38 @@ export interface LifecycleRule {
   readonly objectSizeGreaterThan?: number;
 }
 
+/**
+ * A point in time when an object should transition to a different storage class.
+ */
+export class TransitionMoment {
```

Comment:
Suggestions for better names for this class are welcome :)

Reply:
`TransitionTime` maybe? Then in practice it would look `TransitionTime.at()` and `TransitionTime.after()`